client/src/containers/Annuaire/AnnuaireLecture/components/SearchBarAnnuaire.tsx

```diff
@@ -74,13 +108,81 @@ export const SearchBarAnnuaire = (props: Props) => {
         />
         {props.t("Annuaire.Ville ou département", "Ville ou département")}
       </WhiteButtonContainer>
-      <WhiteButtonContainer>
-        {props.t("Annuaire.Type de structure", "Type de structure")}
-      </WhiteButtonContainer>
-      <WhiteButtonContainer>
+      <Dropdown isOpen={dropdownOpen} toggle={toggle}>
+        <DropdownToggle
+          caret={false}
+          className={
+            typeSelected.length === 0
+              ? "typeButton whiteButton"
+              : "typeButton typeSelected"
+          }
+        >
+          {typeSelected.length === 1
+            ? typeSelected[0]
+            : typeSelected.length > 1
+            ? typeSelected.length + " types"
+            : props.t("Annuaire.Type de structure", "Type de structure")}
+          {typeSelected.length > 0 && (
+            <EVAIcon
+              name="close-circle"
+              fill={colors.blancSimple}
+              onClick={(e: any) => {
+                e.stopPropagation();
+                setTypeSelected([]);
+              }}
+              id="bookmarkBtn"
+              className="ml-10"
+              size={"large"}
+            />
+          )}
+        </DropdownToggle>
+        <DropdownMenu>
+          <DropDownItemContainer>
+            {StructureTypes.map((item: String, key) => {
+              return (
+                <FButton
+                  onClick={() => {
+                    selectType(item);
+                  }}
+                  type="white"
+                  style={
+                    // @ts-ignore
```

Comment:
pourquoi un ts ignore ?

Reply:
sinon j'ai cette erreur Argument of type 'string' is not assignable to parameter of type 'never'
Je ne sais pas comment l'enlever je voulais voir avec toi

Reply:
cest parce que il ne sait pas le type de typeSelected qui est dans ton state. Tu peux typer ton state =useState<string[]>([]) et ca devriat etre bon